src/ReturnTypes/AppMakeHelper.php

```diff
@@ -0,0 +1,50 @@
+<?php
+
+namespace NunoMaduro\Larastan\ReturnTypes;
+
+use NunoMaduro\Larastan\Concerns\HasContainer;
+use PhpParser\Node\Expr\ClassConstFetch;
+use PhpParser\Node\Expr\FuncCall;
+use PhpParser\Node\Expr\MethodCall;
+use PhpParser\Node\Expr\StaticCall;
+use PhpParser\Node\Name\FullyQualified;
+use PhpParser\Node\Scalar\String_;
+use PHPStan\Type\ErrorType;
+use PHPStan\Type\NeverType;
+use PHPStan\Type\ObjectType;
+use PHPStan\Type\Type;
+use Throwable;
+
+final class AppMakeHelper
+{
+    use HasContainer;
+
+    public function resolveTypeFromCall(FuncCall|MethodCall|StaticCall $call): Type
+    {
+        if (count($call->getArgs()) === 0) {
+            return new ErrorType();
+        }
+
+        $expr = $call->getArgs()[0]->value;
+        if ($expr instanceof String_) {
+            try {
+                /** @var object|null $resolved */
+                $resolved = $this->resolve($expr->value);
```

Comment:
it's not really guaranteed that `resolve` will return an object, scalars can be bound into the container using names too and i've seen that in the wild, so it might be worth expanding this code to anticipate more types to be returned in the future